lib/azure/core/http/retry_policy.rb

```diff
@@ -24,7 +24,6 @@ class RetryPolicy < HttpFilter
 
         def initialize(&block)
           @block = block
-          @retry_data = {}
```

Comment:
I made a mistake, sorry.

This initialize code must not delete, because `Azure::Core::Http::HttpRequest#with_filter` uses `retry_data` accessor as hash-initialized.